### PR TITLE
49 venti stack export fluent bit metrics

### LINKED ISSUE #49: [venti-stack] export fluent-bit metrics
[venti-stack] export fluent-bit metrics

### PR DESCRIPTION
**Pull request by jmnote**
#### What this PR does / why we need it

- rearrange fluent-bit values to follow upstream
- export metrics of fluent-bit DaemonSet & lethe-writer
- comment out unused audit function
- luaScripts split into 2 files
- specifies the tags to forward

#### Which issue this PR fixes

- fixes #49 

#### Special notes for your reviewer

Thanks!

#### Checklist

- [x] Chart Version bumped

### CODE REVIEW

File: charts/venti-stack/values.yaml [hunk 1]
```diff
@@ -2,26 +2,31 @@ global:
   imageRepository: ghcr.io/kuoss
 
 fluent-bit:
-  flush: 1
-  logLevel: info
-  metricsPort: 2020
+  # https://github.com/fluent/helm-charts/blob/main/charts/fluent-bit/values.yaml
```
Comment (jmnote) [inline charts/venti-stack/values.yaml:5]: rearrange fluent-bit values to follow upstream

File: charts/venti-stack/values.yaml [hunk 2]
```diff
@@ -2,26 +2,31 @@ global:
...
+  testFramework:
+    enabled: false
+  service:
+    annotations: {}
+    # prometheus.io/path: "/api/v1/metrics/prometheus"
+    # prometheus.io/port: "2020"
+    # prometheus.io/scrape: "true"
```
Comment (jmnote) [inline charts/venti-stack/values.yaml:12]: ready to export fluent-bit metrics

File: charts/venti-stack/values.yaml [hunk 3]
```diff
@@ -2,26 +2,31 @@ global:
...
-        local u = string.sub(r["stageTimestamp"],1,19).."Z"
-        return 1, timestamp, {tag=string.format("audit/audit/%s_%s.log",string.sub(u,1,10),string.sub(u,12,13)),
-        row=string.format("%s %s",u,r["log"])}
-      end
+    # audit.lua: |
+    #   function audit(tag, timestamp, r)
+    #     local u = string.sub(r["stageTimestamp"],1,19).."Z"
+    #     return 1, timestamp, {tag=string.format("audit/audit/%s_%s.log",string.sub(u,1,10),string.sub(u,12,13)),
+    #     row=string.format("%s %s",u,r["log"])}
+    #   end
```
Comment (jmnote) [inline charts/venti-stack/values.yaml:22]: comment out unused audit functions

File: charts/venti-stack/values.yaml [hunk 4]
```diff
@@ -2,26 +2,31 @@ global:
...
+    kube.lua: |
       function kube(tag, timestamp, r)
         local u = os.date("%Y-%m-%dT%H:%M:%SZ")
         return 1, timestamp, {tag=string.format("pod/%s/%s_%s.log",r["kubernetes"]["namespace_name"],string.sub(u,1,10),string.sub(u,12,13)),
         row=string.format("%s[%s|%s|%s] %s",u,r["kubernetes"]["namespace_name"],r["kubernetes"]["pod_name"],r["kubernetes"]["container_name"],r["message"])}
       end
+    host.lua: |
       function host(tag, timestamp, r)
         local u = os.date("%Y-%m-%dT%H:%M:%SZ")
         return 1, timestamp, {tag=string.format("node/%s/%s_%s.log",r["_HOSTNAME"],string.sub(u,1,10),string.sub(u,12,13)),
```
Comment (jmnote) [inline charts/venti-stack/values.yaml:32]: split into 2 files

File: charts/venti-stack/values.yaml [hunk 5]
```diff
@@ -58,34 +63,35 @@ fluent-bit:
...
+          Match_regex         ^(pod|node)
           Remove              tag
 
     outputs: |
       [OUTPUT]
           Name                forward
-          Match               *
+          Match_regex         ^(pod|node)
           Host                {{ .Release.Name }}-lethe
           Port                24224
```
Comment (jmnote) [inline charts/venti-stack/values.yaml:91]: specifies the tags to forward

File: charts/venti-stack/values.yaml [hunk 6]
```diff
@@ -2,26 +2,31 @@ global:
...
-  logLevel: info
-  metricsPort: 2020
+  # https://github.com/fluent/helm-charts/blob/main/charts/fluent-bit/values.yaml
+  testFramework:
+    enabled: false
+  service:
+    annotations:
+      prometheus.io/path: "/api/v1/metrics/prometheus"
+      prometheus.io/port: "2020"
+      prometheus.io/scrape: "true"
```
Comment (jmnote) [inline charts/venti-stack/values.yaml:12]: export metrics of fluent-bit DaemonSet

File: charts/venti-stack/values.yaml [hunk 7]
```diff
@@ -116,7 +122,10 @@ lethe:
 
   service:
     enabled: true
-    annotations: {}
+    annotations:
+      prometheus.io/path: "/api/v1/metrics/prometheus"
+      prometheus.io/port: "2020"
+      prometheus.io/scrape: "true"
```
Comment (jmnote) [inline charts/venti-stack/values.yaml:128]: export metrics of fluent-bit (lethe-writer)

File: docs/venti-stack.values.test.yaml
Comment (jmnote) [inline docs/venti-stack.values.test.yaml:1]: not important (values.yaml for test)

File: charts/venti-stack/templates/lethe/service.yaml [hunk 8]
```diff
@@ -38,18 +38,14 @@ spec:
     {{- if .Values.lethe.service.nodePort }}
       nodePort: {{ .Values.lethe.service.nodePort }}
     {{- end }}
-    - name: http-input
-      port: 9880
+    - name: metrics
+      port: 2020
       protocol: TCP
-      targetPort: 9880
```
Comment (jmnote) [inline charts/venti-stack/templates/lethe/service.yaml:44]: comment out unused port

**Review by moonek — APPROVED** [review]

**Pull request merged**